frame/contracts/src/wasm/code_cache.rs

```diff
@@ -190,7 +191,10 @@ pub fn reinstrument<T: Config>(
 	let original_code =
 		<PristineCode<T>>::get(&prefab_module.code_hash).ok_or(Error::<T>::CodeNotFound)?;
 	let original_code_len = original_code.len();
-	prefab_module.code = prepare::reinstrument_contract::<T>(original_code, schedule)?;
+	prefab_module.code = prepare::reinstrument_contract::<T>(&original_code, schedule)
+		.map_err(|_| <Error<T>>::CodeRejected)?
```

Comment:
`check_and_instrument` could be changed to return a `DispatchError` to prevent this `map_err(|_| <Error<T>>::CodeRejected)` here and in `do_preparation`

Reply:
We can't throw away the string describing the error here because it is used in `internal_instantiate` in order to write it to the debug buffer.